Automattic/woocommerce-payments
# Storing transaction fee in order metadata

Fixes #3257

#### Changes proposed in this Pull Request

- The `application_fee_amount` is stored in order metadata as `_wcpay_transaction_fee`
- Introduced a function `attach_transaction_fee_to_order` in `WC_Payments_Order_Service`
- Saves at different points to cover all checkout flows
            -   `process_payment_for_order` (Regular Order)
            -  `process_redirect_payment` ( UPE)
            -  `process_webhook_payment_intent_succeeded` (Sometimes the redirect processing is not successful and charge data is found missing on order)
-  Displays the Transaction Fee on Order Screen

##### Screenshot
![image](https://user-images.githubusercontent.com/32092402/227422641-cf186e57-3ea8-4b1d-869c-87131a9edf23.png)

#### Testing instructions
- Checkout using different flows - Classic checkout, Block checkout, UPE, non-UPE, other Payment Methods, using saved card
- Verify that the fee charged by Stripe is same as the value saved into order metadata with key `_wcpay_transaction_fee`
- Verify that the same value is displayed on the Order Edit screen.

-------------------


**Post merge**



- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-5.7.0#test-order-transaction-fee-display
